File: final_assignment_front/lib/features/model/user_management.dart
class UserManagement {
  int? userId;
  String? username;
  String? password;
  String? contactNumber;
  String? email;
  String? status;
  DateTime? createdTime;
  DateTime? modifiedTime;
  String? remarks;
  String? idempotencyKey;

  UserManagement({
    this.userId,
    this.username,
    this.password,
    this.contactNumber,
    this.email,
    this.status,
    this.createdTime,
    this.modifiedTime,
    this.remarks,
    this.idempotencyKey,
  });

  @override
  String toString() {
    return 'UserManagement[userId=$userId, username=$username, password=$password, contactNumber=$contactNumber, email=$email, status=$status, createdTime=$createdTime, modifiedTime=$modifiedTime, remarks=$remarks, idempotencyKey=$idempotencyKey]';
  }

  factory UserManagement.fromJson(Map<String, dynamic> json) {
    return UserManagement(
      userId: json['userId'] as int?,
      username: json['username'] as String?,
      password: json['password'] as String?,
      contactNumber: json['contactNumber'] as String?,
      email: json['email'] as String?,
      status: json['status'] as String?,
      createdTime: json['createdTime'] != null ? DateTime.parse(json['createdTime'] as String) : null,
      modifiedTime: json['modifiedTime'] != null ? DateTime.parse(json['modifiedTime'] as String) : null,
      remarks: json['remarks'] as String?,
      idempotencyKey: json['idempotencyKey'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> json = {};
    if (userId != null) json['userId'] = userId;
    if (username != null) json['username'] = username;
    if (password != null) json['password'] = password;
    if (contactNumber != null) json['contactNumber'] = contactNumber;
    if (email != null) json['email'] = email;
    if (status != null) json['status'] = status;
    if (createdTime != null) json['createdTime'] = createdTime!.toIso8601String();
    if (modifiedTime != null) json['modifiedTime'] = modifiedTime!.toIso8601String();
    if (remarks != null) json['remarks'] = remarks;
    if (idempotencyKey != null) json['idempotencyKey'] = idempotencyKey;
    return json;
  }

  static List<UserManagement> listFromJson(List<dynamic> json) {
    return json.map((value) => UserManagement.fromJson(value as Map<String, dynamic>)).toList();
  }

  static Map<String, UserManagement> mapFromJson(Map<String, dynamic> json) {
    var map = <String, UserManagement>{};
    if (json.isNotEmpty) {
      json.forEach((key, value) => map[key] = UserManagement.fromJson(value));
    }
    return map;
  }

  static Map<String, List<UserManagement>> mapListFromJson(Map<String, dynamic> json) {
    var map = <String, List<UserManagement>>{};
    if (json.isNotEmpty) {
      json.forEach((key, value) => map[key] = UserManagement.listFromJson(value));
    }
    return map;
  }
}

File: final_assignment_backend_go/project/internal/domain/user_management.go
package domain

import (
	"time"

	"gorm.io/gorm"
)

// UserManagement maps traffic.sys_user and emits camelCase JSON.
type UserManagement struct {
	UserID          int            `gorm:"column:user_id;primaryKey;autoIncrement" json:"userId"`
	Username        string         `gorm:"column:username" json:"username"`
	Password        string         `gorm:"column:password" json:"-"`
	RealName        string         `gorm:"column:real_name" json:"realName"`
	ContactNumber   string         `gorm:"column:contact_number" json:"contactNumber"`
	Email           string         `gorm:"column:email" json:"email"`
	Department      string         `gorm:"column:department" json:"department"`
	Position        string         `gorm:"column:position" json:"position"`
	Status          string         `gorm:"column:status" json:"status"`
	LastLoginTime   *time.Time     `gorm:"column:last_login_time" json:"lastLoginTime"`
	CreatedTime     *time.Time     `gorm:"column:created_at" json:"createdTime"`
	ModifiedTime    *time.Time     `gorm:"column:updated_at" json:"modifiedTime"`
	Remarks         string         `gorm:"column:remarks" json:"remarks"`
	DeletedAt       gorm.DeletedAt `gorm:"column:deleted_at;index" json:"-"`
}

func (UserManagement) TableName() string {
	return "sys_user"
}
